### DISCUSSION TITLE
ListCost is less than x_OnDemandCost

### DISCUSSION THREAD

**Discussion by lundejd** [microsoft/finops-toolkit#739] [General]
I have a freshly installed FinOps hub and am seeing some strange results. Perhaps this is normal and I'm interpreting them wrong - I did also see there were some issues with some of these columns.

So if ListCost is "Derived if not available. List (or retail) cost without any discounts applied." I would think this is the public pay-as-you-go cost and would show as the highest cost of all the various columns.

x_OnDemandCost is "Cost based on UnitPrice (with negotiated discounts applied, but without commitment-based discounts). Calculated as Quantity multiplied by UnitPrice." 

NegotiatedCostSavings is "Amount saved after negotiated discounts are applied but excluding commitment discounts. Formula: ListCost - x_OnDemandCost." so I would think ListCost is highest, x_OnDemandCost is then ListCost minus our EA discount, and the NegotiatedCostSavings should be the difference.

With an x_OnDemandCost of $1,352,000 I see a ListCost of $214,180 and a x_NegotiatedCostSavings close to $50k.

Let's just say our EA discount is 25% I would expect ListCost to be closer to $1,802,700 and x_NegotatedCostSavings to be around $450k. Is this the right assumption?

Is this still data that has a bug? I saw references in a few other discussions that it was looked at and fixed so I though I'd bring it forward.

**Comment by flanakin** [comment]
Cost Management does not include ListCost for every row, so you won't be able to sum ListCost alone. This is why x_NegotiatedCostSavings is not a simple delta between x_OnDemandCost and EffectiveCost.

Open the Commitment discounts report and look at the `DQ` page for a little insight into which rows are impacted by this. Going forward, we're thinking about backfilling the missing data to at least give you a more representative picture of savings from what we can identify, but it will still be partial until Cost Management provides the missing data.

As a note, we're also working on a proof of concept to merge cost data with prices to fill these gaps. That is progressing well but is likely a couple months away. This is not the scenario we were originally targeting, but we can likely reuse what we have to cover this as well.

As a side note, assuming this is due to the data missing in Cost Management, you can file a support request to raise this issue with them. They may or may not be able to resolve it, but it will at a minimum raise more awareness of the gap and increase the priority to getting it resolved.

**Comment by lundejd** [comment]
Ok this makes sense. I will open a ticket.

Seems like one could open a pay-as-you-go list price account and pull the costs in as a measure. (only downfall might be products that are preview that don’t show up in the pay-as-you-go tenant)

Even easier if x_OnDemandCost is reliable and correct we could also just say NewListCost = [x_OnDemandCost] x 1.Y where Y is your EA discount percent.

From: Michael Flanakin ***@***.***>
Date: Thursday, June 13, 2024 at 5:48 PM
To: microsoft/finops-toolkit ***@***.***>
Cc: John Lundell ***@***.***>, Author ***@***.***>
Subject: Re: [microsoft/finops-toolkit] ListCost is less than x_OnDemandCost (Discussion #739)

Cost Management does not include ListCost for every row, so you won't be able to sum ListCost alone. This is why x_NegotiatedCostSavings is not a simple delta between x_OnDemandCost and EffectiveCost.

Open the Commitment discounts report and look at the DQ page for a little insight into which rows are impacted by this. Going forward, we're thinking about backfilling the missing data to at least give you a more representative picture of savings from what we can identify, but it will still be partial until Cost Management provides the missing data.

As a note, we're also working on a proof of concept to merge cost data with prices to fill these gaps. That is progressing well but is likely a couple months away. This is not the scenario we were originally targeting, but we can likely reuse what we have to cover this as well.

As a side note, assuming this is due to the data missing in Cost Management, you can file a support request to raise this issue with them. They may or may not be able to resolve it, but it will at a minimum raise more awareness of the gap and increase the priority to getting it resolved.
